homeassistant/components/slide_local/cover.py

```diff
@@ -0,0 +1,152 @@
+"""Support for Slide covers."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    PLATFORM_SCHEMA as COVER_PLATFORM_SCHEMA,
+    CoverDeviceClass,
+    CoverEntity,
+)
+from homeassistant.const import (
+    CONF_API_VERSION,
+    CONF_HOST,
+    CONF_PASSWORD,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPENING,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import SlideConfigEntry
+from .const import CONF_INVERT_POSITION, DEFAULT_OFFSET
+from .coordinator import SlideCoordinator
+from .entity import SlideEntity
+
+COVER_PLATFORM_SCHEMA = COVER_PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_HOST): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_INVERT_POSITION, default=False): cv.boolean,
+        vol.Optional(CONF_API_VERSION, default=2): cv.byte,
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SlideConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up cover(s) for Slide platform."""
+
+    _LOGGER.debug(
+        "Initializing Slide cover '%s'",
+        entry.data[CONF_HOST],
+    )
+
+    coordinator: SlideCoordinator = entry.runtime_data
+
+    if coordinator.data.get("mac") == "":
+        _LOGGER.error(
+            "Unable to setup Slide Local '%s', the MAC is missing in the slide response",
+            entry.data[CONF_HOST],
+        )
+        return
+
+    async_add_entities(
+        [
+            SlideCoverLocal(
+                coordinator,
+                entry,
+            )
+        ]
+    )
+
+    _LOGGER.debug("Setup Slide '%s' successful", entry.data[CONF_HOST])
+
+
+class SlideCoverLocal(SlideEntity, CoverEntity):
+    """Representation of a Slide Local API cover."""
+
+    _attr_assumed_state = True
+    _attr_device_class = CoverDeviceClass.CURTAIN
+
+    def __init__(
+        self,
+        coordinator,
+        entry: SlideConfigEntry,
+    ) -> None:
+        """Initialize the cover."""
+        super().__init__(coordinator)
+
+        self._attr_name = coordinator.data["device_name"]
+        self._invert = entry.data[CONF_INVERT_POSITION]
+        self._attr_unique_id = f"{coordinator.data["mac"]}-cover"
```

Comment:
We don't need to add `-cover` to it, since IDs are unique per platform per integration